src/josepy/util.py

```diff
@@ -99,12 +85,12 @@ def __eq__(self, other):
         else:
             return NotImplemented
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return '<{0}({1!r})>'.format(self.__class__.__name__, self._wrapped)
 
-    def public_key(self):
+    def public_key(self) -> 'ComparableKey':
         """Get wrapped public key."""
-        return self.__class__(self._wrapped.public_key())
+        return self.__class__(self._wrapped.public_key())  # type: ignore[union-attr]
```

Comment:
This is a bit of a nitpick, but I think this is a weird function here and in `ComparableECKey` below. I think `self._wrapped` will blow up if `self._wrapped` is either public key type right? If that's the behavior we want (which seems fine to me personally), why not check for a `public_key` attribute and error if it doesn't exist. Alternatively, if we want to support public keys like we try to in `ComparableECKey` maybe we could use the private numbers approach (or just return the public key directly)?

My hope is either approach would allow us to remove the `type: ignore` lines. What do you think?

Reply:
Yes with `private_numbers` and self return in case of a public key, we can have a consistent behavior. I put a proposition in the code which allows to merge the `public_key()`  method in `ComparableKey`.